tools/rewriter/SourceRewriter.cpp

```diff
@@ -145,7 +149,7 @@ static bool ignore_file(const Filename &filename) {
   if (is_empty) {
     return false;
   }
-  return !filename.starts_with(RootDirectory);
+  return !filename.starts_with(OutputDirectory) && !filename.starts_with(RootDirectory);
```

Comment:
I agree my change was wrong, but I think this still mishandles the case of root directory/not output directory. We shouldn't encounter this case because it means we're either trying to annotate an original source file in-place or picking up one of the original (non-system) headers. I think it'd be clearer to just ignore files not in the output directory and assert that if it's not in the output directory it's also not in the root directory.

Reply:
You're right, done. I had thought the `RootDirectory` check was necessary but in testing it is never encountered and as you observe shouldn't be. I've added an error if we do somehow encounter that case in the future.